crypto/src/main/java/net/consensys/cava/crypto/mikuli/PrivateKey.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.cava.crypto.mikuli;
+
+/**
+ * This class represents a private key on G2
```

Comment:
I hate to even answer this question honestly bc I can see this getting re-engineered. Basically, the code I based this on originally derived private keys from the elliptic curve subgroup G1 and hash the message to the G1 elliptic curve subgroup. Public keys and the generating point we're on G2.  I switched how it worked so it would meet spec. Admittedly, the explanation could be better, but it does mean something to people familiar with BLS signatures

Reply:
It was a rhetorical question :) I'd just change it to say "A Mikuli Private Key". Although what is the difference between Mikuli and BLS12381?